### PR TITLE
[npm-profile] add missing APIs for Login and Account Creation

### PR DESCRIPTION
**Pull request by LonghronShen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/npm/npm-profile#api>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## npm-profile ([<kbd>unpkg</kbd>](https://unpkg.com/browse/npm-profile@latest/))
was missing the following properties:
1. loginWeb
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1b7821f91504c9eb552fa51ebae08c859572e175
</p>

File: types/npm-profile/index.d.ts [hunk 1]
```diff
@@ -40,6 +40,47 @@ export function createToken(
     options?: Options,
 ): Promise<Token>;
 
+/**
+ * Creates a new user on the server along with a fresh bearer token for future authentication as this user.
+ * This is what you see as an authToken in an .npmrc.
+ * @async
+ */
+export function adduser(opener: (url: string) => Promise<void>, prompter: (creds: ProfileAuthCredentials) => Promise<ProfileAuthCredentials>, opts: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:48]:
```suggestion
export function adduser(opener: (url: string) => Promise<void>, prompter: (creds: ProfileAuthCredentials) => Promise<ProfileAuthCredentials>, opts?: Options): Promise<ProfileAuthToken>;
```
this is a notation used here for optional parameter (I assume this is the intent here)

File: types/npm-profile/index.d.ts [hunk 2]
```diff
@@ -40,6 +40,47 @@ export function createToken(
...
+
+/**
+ * Tries to login using new web based login, if that fails it falls back to using the legacy CouchDB APIs.
+ * @async
+ */
+export function login(opener: (url: string) => Promise<void>, prompter: (creds: ProfileAuthCredentials) => Promise<ProfileAuthCredentials>, opts: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:54]:
```suggestion
export function login(opener: (url: string) => Promise<void>, prompter: (creds: ProfileAuthCredentials) => Promise<ProfileAuthCredentials>, opts?: Options): Promise<ProfileAuthToken>;
```

File: types/npm-profile/index.d.ts [hunk 3]
```diff
@@ -40,6 +40,47 @@ export function createToken(
...
+
+/**
+ * Tries to login using new web based login, if that fails it falls back to using the legacy CouchDB APIs.
+ * @async
+ */
+ export function loginWeb(opener: (url: string) => Promise<void>, opts: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:60]:
```suggestion
 export function loginWeb(opener: (url: string) => Promise<void>, opts?: Options): Promise<ProfileAuthToken>;
```

File: types/npm-profile/index.d.ts [hunk 4]
```diff
@@ -40,6 +40,47 @@ export function createToken(
...
+
+/**
+ * Tries to create a user new web based login,
+ * if that fails it falls back to using the legacy CouchDB APIs.
+ * @async
+ */
+export function adduserWeb(opener: (url: string) => Promise<void>, opts: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:67]:
```suggestion
export function adduserWeb(opener: (url: string) => Promise<void>, opts?: Options): Promise<ProfileAuthToken>;
```

File: types/npm-profile/index.d.ts [hunk 5]
```diff
@@ -40,6 +40,47 @@ export function createToken(
...
+
+/**
+ * Creates a new user on the server along with a fresh bearer token for future authentication as this user.
+ * This is what you see as an authToken in an .npmrc.
+ * @async
+ */
+export function adduserCouch(username: string, email: string, password: string, opts?: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:74]:
```suggestion
export function adduserCouch(username: string, email: string, password: string, opts?: Options): Promise<ProfileAuthToken>;
```

File: types/npm-profile/index.d.ts [hunk 6]
```diff
@@ -40,6 +40,47 @@ export function createToken(
...
+
+/**
+ * Logs you into an existing user. Does not create the user if they do not already exist.
+ * Logging in means generating a new bearer token for use in future authentication.
+ * This is what you use as an authToken in an .npmrc.
+ * @async
+ */
+export function loginCouch(username: string, email: string, password: string, opts?: Options | {}): Promise<ProfileAuthToken>;
```
Comment (peterblazejewicz) [inline types/npm-profile/index.d.ts:82]:
```suggestion
export function loginCouch(username: string, email: string, password: string, opts?: Options): Promise<ProfileAuthToken>;
```

File: types/npm-profile/index.d.ts (hunk 1 above)
Comment (LonghronShen) [reply types/npm-profile/index.d.ts]: I think your suggestion is right, yes, this is an optional parameter.

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@LonghronShen congrats on your first contribution here!

**Comment by LonghronShen** [comment]
Thank you for your time and patience~

**Comment by LonghronShen** [comment]
Ready to merge

**Pull request merged**